src/store/DevfileRegistries/selectors.ts

```diff
@@ -14,22 +14,29 @@ import { createSelector } from 'reselect';
 import { AppState } from '../';
 import match from '../../services/helpers/filter';
 
-const selectState = (state: AppState) => state.devfileRegistries;
+const selectState = (state: AppState) => ({ devfileRegistries: state.devfileRegistries, workspacesSettings: state.workspacesSettings });
```

Comment:
I think of importing the `selectWorkspacesSettingsState` or just creating this state so you don't need to modify other selectors but only `selectRegistriesMetadata`:

```typescript

import { selectWorkspacesSettingsState } from '../Workspaces/Settings/selectors';
// or
const selectWorkspacesSettingsState = (state: AppState) => state.workspacesSettings;

const selectState = (state: AppState) => state.devfileRegistries;

export const selectRegistriesMetadata = createSelector(
  selectState,
  selectWorkspacesSettingsState,
  (devfileRegistriesState, workspacesSettingsState) => {
    const registriesMetadata = Object.values(devfileRegistriesState.registries)
      .map(registryMetadata => registryMetadata.metadata || []);
    const metadata = mergeRegistriesMetadata(registriesMetadata);
    const cheDevworkspaceEnabled = workspacesSettingsState.settings['che.devworkspaces.enabled'] === 'true';
    if (cheDevworkspaceEnabled) {
      return filterDevfileV2Metadata(metadata);
    } else {
      return metadata;
    }

  }
);
```

Reply:
yes it looks far better